Custom Color Themes

arcanda3 opened the pull request:
Integrated custom color themes using the [Material Theme Builder](https://www.figma.com/community/plugin/1034969338659738588/material-theme-builder) plugin in Figma to create custom color schemes for purple, pink, yellow, green, blue, and brown. The default theme is purple, which closely mirrors the previous default theme from React Native Paper. 

The theme JSON files are stored in `constants/themes/` and are parsed by the `ThemeHandler`, which transforms them into the format expected by React Native Paper. Users can toggle between different themes using the `ThemeSelector` component located in the settings page. 

When the theme is changed, it is saved to the settings table in the database to ensure the selected theme persists when the app is closed and reopened. Additionally, I updated the iOS `BlurTabBarBackground` to use default, as the previous mode didn't align well with the new themed background colors.

<img src="https://github.com/user-attachments/assets/6923a3f9-c781-43a5-ad0c-43464d870709" width="150"> <img src="https://github.com/user-attachments/assets/273be11c-5d41-4687-b8f4-7283b92c4935" width="150"> <img src="https://github.com/user-attachments/assets/f0330442-30ff-49e4-934f-04c18478032f" width="150"> <img src="https://github.com/user-attachments/assets/c476fe36-5eb6-437e-aa03-df0d7e03b1cc" width="150">

s-egge commented:
It's looking awesome so far! I did notice that the dialogs aren't changing color with the theme properly. 
![image](https://github.com/user-attachments/assets/7b0abaee-ee1a-4686-b615-4b3ec9478c2f)

I was actually testing theming on a different project over the weekend and ran in to the same issue and realized that the Material Theme Builder doesn't have the "elevation" object in their theme that React Native Paper uses for Dialogs and some buttons. I had forgotten to update the GH Issue here, but you can build color schemes right in React Native Paper's theming page, I did a quick test and was able to update the dialogs by just making a color scheme from the primary color in the `green.json` file and replacing the "light" and "dark" object values. 
![image](https://github.com/user-attachments/assets/0c1db5ee-d49d-4498-b1c2-b3edffb6ed40)

It doesn't have the different types of contrast that the MD Builder does though, so if we wanted to add that custom option in the future for medium and high contrasts, we'd probably need to manually convert the elevation values based on RNP's description of them:
![image](https://github.com/user-attachments/assets/10d7dde7-3335-4915-9828-917caf5f411e)

Once that's fixed it should be good to go though!

arcanda3 commented:
> It's looking awesome so far! I did notice that the dialogs aren't changing color with the theme properly. ![image](https://private-user-images.githubusercontent.com/102624422/417201346-7b0abaee-ee1a-4686-b615-4b3ec9478c2f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA1OTA2NjQsIm5iZiI6MTc0MDU5MDM2NCwicGF0aCI6Ii8xMDI2MjQ0MjIvNDE3MjAxMzQ2LTdiMGFiYWVlLWVlMWEtNDY4Ni1iNjE1LTRiM2VjOTQ3OGMyZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIyNlQxNzE5MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMTNhMDhiNTYwYTI3ZDhhMWU4YjMzOTRhMGJlZjUxMzZhYzRkNDg3MTNhY2ZhZTlhYjNhNGU0ZjU3MWZkYjY2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9._vrIksK7OCnrHqvVlqzkXe6OEt1rLXNwyMqYIK_qVMo)
> 
> I was actually testing theming on a different project over the weekend and ran in to the same issue and realized that the Material Theme Builder doesn't have the "elevation" object in their theme that React Native Paper uses for Dialogs and some buttons. I had forgotten to update the GH Issue here, but you can build color schemes right in React Native Paper's theming page, I did a quick test and was able to update the dialogs by just making a color scheme from the primary color in the `green.json` file and replacing the "light" and "dark" object values. ![image](https://private-user-images.githubusercontent.com/102624422/417205351-0c1db5ee-d49d-4498-b1c2-b3edffb6ed40.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA1OTA2NjQsIm5iZiI6MTc0MDU5MDM2NCwicGF0aCI6Ii8xMDI2MjQ0MjIvNDE3MjA1MzUxLTBjMWRiNWVlLWQ0OWQtNDQ5OC1iMWMyLWIzZWRmZmI2ZWQ0MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIyNlQxNzE5MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zZGE1NGYxZGU0NzZiM2E2Y2Y5N2JjNDFmZGM2MjJjYzQyNmZlYjYxMjIwZjI2MTc0ZTdkNzMzMzFkNTcyYTAyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.bJfDTRr36fchTQv4tGBai0NfYp9YGBfusqrmx_MdMoA)
> 
> It doesn't have the different types of contrast that the MD Builder does though, so if we wanted to add that custom option in the future for medium and high contrasts, we'd probably need to manually convert the elevation values based on RNP's description of them: ![image](https://private-user-images.githubusercontent.com/102624422/417205874-10d7dde7-3335-4915-9828-917caf5f411e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA1OTA2NjQsIm5iZiI6MTc0MDU5MDM2NCwicGF0aCI6Ii8xMDI2MjQ0MjIvNDE3MjA1ODc0LTEwZDdkZGU3LTMzMzUtNDkxNS05ODI4LTkxN2NhZjVmNDExZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIyNlQxNzE5MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04YzIyZmE4ODY5NmZlNDJiNThkOGJkNzRiNzFiODk0ZjQwMTg1MTE2YTk1ZTg1ODg2OGE2ZmQwYTMzMzc3NWU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.2fZXxuDBbFxvVUWUhXarrECnk12OEqfCy_leJji_9ZM)
> 
> Once that's fixed it should be good to go though!

Thanks for letting me know! Using the react native paper theming objects from the site introduced a bug with the coloring for the tab bar icons on iOS that I'm working on fixing right now.

arcanda3 commented:
Should be fixed now! There were some discrepancies with react native paper vs MD3 theming formats. I converted all the color values to rgb/rgba to preserve any transparent colors generated by the react native paper theme maker and added the color types that were missing in the MD3 format to the theme color files.

s-egge reviewed (approved):
It looks great!

Blake-Hua reviewed (approved):
Tested theme colors! The colors look very nice and I didn't run into any issues
